Packs/Cisco-umbrella-cloud-security/Integrations/CiscoUmbrellaCloudSecurityv2/CiscoUmbrellaCloudSecurityv2.py

```diff
@@ -0,0 +1,890 @@
+    """
+    Determine whether the request arguments are for a GET or LIST request.
+
+    Args:
+        get_args (list): GET request arguments.
+        list_args (list): LIST request arguments.
+
+    Raises:
+        ValueError: In case the user has entered both GET and LIST arguments, raise an error.
+
+    Returns:
+        bool: True if the arguments are for a GET request, False otherwise.
+    """
+    is_get_request = any(get_args)
+    is_list_request = any(list_args)
+
+    if is_get_request and is_list_request:
+        raise ValueError('GET and LIST arguments can not be supported simultaneously.')
+
+    return is_get_request
+
+
+@logger
+def get_json_table(obj: OptionalDictOrList, json_transformer: JsonTransformer) -> OptionalDictOrList:
+    """Convert a dict or list to a table.
+
+    Args:
+        obj (OptionalDictOrList): The dict or list to convert.
+        json_transformer (JsonTransformer): The json transformer to use.
+
+    Returns:
+        OptionalDictOrList: The converted dict or list, otherwise returns obj if nothing was inserted.
+    """
+    if not obj:
+        return obj
+
+    def get_json_table_for_dict(data: dict[str, Any]) -> dict[str, Any]:
+        """Convert a dict to a table.
+
+        Args:
+            data (dict[str, Any]): The dict to convert.
+
+        Returns:
+            dict[str, Any]: The converted dict.
+        """
+        return {
+            transformed_data[1]: transformed_data[2]
+            for transformed_data in json_transformer.json_to_path_generator(data)
+        }
+
+    if isinstance(obj, list):
+        return [get_json_table_for_dict(o) for o in obj]
+
+    return get_json_table_for_dict(obj)
+
+
+@logger
+def handle_pagination(
+    list_command: Callable[..., dict[str, Any]],
+    limit: int,
+    *args,
+    page_size: int | None = None,
+    page: int | None = None,
+    **kwargs,
+) -> tuple[list[dict[str, Any]], list[dict[str, Any]] | dict[str, Any]]:
+    """Handles pagination for the given list_command.
+
+    Args:
+        list_command (Callable[..., dict[str, Any]]): Callable that returns data containing destinations.
+        limit (int): The maximum number of items to return.
+        page (int | None, optional): The page number to start from.
+            Defaults to None.
+        page_size (int | None, optional): The number of items to return per page.
+            Defaults to None.
+        *args: Arguments passed down by the CLI to configure the request.
+        **kwargs: Keyword arguments passed down by the CLI to configure the request.
+
+    Returns:
+        tuple[list[dict[str, Any]], list[dict[str, Any]] | dict[str, Any]]: A tuple containing the list of
+        items and raw responses, or a single item and raw response if page is None.
+    """
+    remaining_items = page_size or limit
+
+    if page:
+        remaining_items = min(remaining_items, API_LIMIT)
+        raw_response = list_command(*args, page=page, limit=remaining_items, **kwargs)
+
```

Comment:
Do you want me to erase this? Why?

Reply:
The following line is still part of this logical block, no reason to split them vertically

Reply:
Do you mean combine the `if` block logic within the loop? 
I've separated them so there will be a scope for manual pagination and another for automatic.
It is possible to combine them, but it will be a bit messy and we will have to repeat some of the logic within each loop.
In the automatic we must call with `API_LIMIT` all the time so we can go through the pages in an organized order (the API itself implements pagination) so we can mimic `limit` and `offset` some how.

Reply:
Removed line.